resolver/resolver.go

```diff
@@ -16,7 +15,7 @@ import (
 
 	"github.com/emicklei/go-restful"
 	"github.com/mesos/mesos-go/detector"
-	_ "github.com/mesos/mesos-go/detector/zoo"
+	_ "github.com/mesos/mesos-go/detector/zoo" // Registers the ZK detector
```

Comment:
Yes it does: https://github.com/mesos/mesos-go/blob/master/detector/zoo/plugin.go#L7

Side effects... Side effects everywhere! :)